File: tablet/Resources/menu/PsychosisMenu.js
// PsychosisMenu.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function PsychosisMenu() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            UICONSTANTS.CHANGE_PATIENT_MENU_LINE,
            ALLTASKS.TASKLIST.CAPE42,
            ALLTASKS.TASKLIST.CAPS,
            ALLTASKS.TASKLIST.CGISCH,
            ALLTASKS.TASKLIST.PANSS,
            ALLTASKS.TASKLIST.ICD10_SCHIZOPHRENIA,
            ALLTASKS.TASKLIST.ICD10_SCHIZOTYPAL,
        ],
        self = new MenuWindow({
            title: L('menutitle_psychosis'),
            icon: UICONSTANTS.ICON_MENU_PSYCHOSIS,
            data: data,
        });
    return self;
};

File: tablet_qt/menu/psychosismenu.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

#include "psychosismenu.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"
#include "tasks/cape42.h"
#include "tasks/caps.h"
#include "tasks/cgisch.h"
#include "tasks/icd10schizophrenia.h"
#include "tasks/icd10schizotypal.h"
#include "tasks/panss.h"

PsychosisMenu::PsychosisMenu(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_PSYCHOSIS))
{
}

QString PsychosisMenu::title() const
{
    return tr("Psychosis");
}

void PsychosisMenu::makeItems()
{
    m_items = {
        MAKE_CHANGE_PATIENT(m_app),
        MAKE_TASK_MENU_ITEM(Cape42::CAPE42_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Caps::CAPS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(CgiSch::CGISCH_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Panss::PANSS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Icd10Schizophrenia::ICD10SZ_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Icd10Schizotypal::ICD10SZTYPAL_TABLENAME, m_app),
    };
}
